program/src/processor/sweep_fees.rs

```diff
@@ -89,7 +89,7 @@ pub(crate) fn process(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramR
     check_accounts(program_id, &market_state, &accounts)?;
     check_metadata_account(accounts.token_metadata, &market_state.base_mint)?;
 
-    if market_state.accumulated_fees == 0 {
+    if market_state.accumulated_fees == 0 && market_state.accumulated_royalties == 0 {
```

Comment:
I think it would make sense to wrap the transfers with these if branches, instead of erroring. This way, if a new transfer is introduced in this function, it would not early abort (which would also mitigate the earlier issue).